app/vlselect/logsql/logsql.go

```diff
@@ -473,6 +509,8 @@ type tailProcessor struct {
 
 	perStreamRows  map[string][]logRow
 	lastTimestamps map[string]int64
+	resultRows     []logRow
```

Comment:
This optimization complicates the code, while it doesn't bring significant benefits for this particular case with live tailing. So it is unlikely this optimization should be added here.